project/jasonchen1221/README.md

```diff
@@ -0,0 +1,115 @@
+A fundamental problem to floorplanning/placement lies in the representation of geometric relationship among modules. As the problem size becomes larger and larger, the representation profoundly affects the operations of modules and the complexity of floorplan/placement design process. Its is that desired to develop an efficient, flexiblem and effective representation of geometric relationship for floorplan/placement designs.
+
+In [1], Lin proposed an Hierarchical B*-tree(HB*-tree) data structure and algorithms that can efficiently handle the symmetric and non-symmetric modules when doing the floorplan/placement. It is based on concept of B*-tree which was proposed by Chang in [2]. Operations on HB*-tree nodes to do the packing and perturbation for analog placement only need time complexity O(lgn) and O(n), respectively. It is very efficient compare to other methods mentioned in [2]. 
+
+Hence, I will use HB*-tree as the data structure to handle the modules. And the algorithms are as below:
+1. HB*-tree
+    1. Insertion
+    1. Deletion
+    1. Contour nodes update
+2. HB*-tree perturbation in floorplan process
+    1. Op1: Rotate a module
+    2. Op2: Move a node to another place
+    3. Op3: Swap two nodes
+    4. Op4: Change a representative
+    5. Op5: Convert a symmetry type
+
+## Perspective Users
+Analog circuit designer, layout designer.
+
+
+## System Architecture
+### Flow Chart
+![This is a alt text.](/img/workflow.png "flow chart")
+
+1. Inputs
+    1. Alpha value
+    2. Symmetry informations
+    3. Nets informations
+    4. Blocks informations
+2. Outputs
+    1. Text file: Coordinates informations for each blocks after placement
+    2. Picture : placement topology and HPWL analysis
+
+
+## API Description
+1. Use pytest to test the correctness of implementation of HB*-tree, also plot the HB*-tree packing results using Matplotlib in python
+2. Use pybind11 as an interface to set the input files, alpha value, and simulated annealing time.
+
+## Engineering Infrastructure
+1. Compilation & Execution: use make to compile, run, plot, clean
+2. Version control: use git
+
+## Schedule
+**Before Week 1:**
+* Planning phase (from 10/10 to 10/31): 
+    1. Study related paper and works
+    2. Prepare proposal
+
+**Week 1 (10/31):**
+* Implement parser and basic data structures(Block)
+* Start the presentation skeleton. Contents will be added to the presentation along with the code development.
+
+
+**Week 2 (11/7):**
+* Develop HB*-tree data structure and related operations
+    1. Node insertion, deletion
+    2. contour line
+    3. packing 
+* Construcy the pytest structure to test the correctness of HB*-tree
+
+**Week 3 (11/14):**
+* Develop HB*-tree data structure and related operations(cont.)
+* Using Matplotlib to implement the Visualization of the HB*-tree packing results.(Also for viewing perturbation operations results in later weeks)
+
+**Week 4 (11/21):**
+* Refactoring week (and accommodation of overflown work).
```

Comment:
It is about reasonable time to develop the code for a search tree in 2 business weeks.  It might be very intensive coding efforts.  If it takes too much time, you can reduce the scope.  Only do some of the operations for the search.